internal/db/collection.go

```diff
@@ -520,6 +525,11 @@ func (c *collection) update(
 	ctx context.Context,
 	doc *client.Document,
 ) error {
+	err := c.setEmbedding(ctx, doc, false)
```

Comment:
question: Is there a particular reason why the first thing the `create/update` call does is generate the embeddings? in the `update` case, it gets the embeddings before the ACP check is even done to determine if a mutation is allowed? It could even be in the `save` func deeper in the call graph instead of duplicated in the `create` and `update` paths.

Reply:
It can't be in the `c.save` method because save is used with both create and update. It might be ok for update but on create, the `c.setEmbedding` bust be call before other actions that are create specific like ensuring the docID is unique and such.

So for create I'll leave it where it is and for update I'll move it bellow the ACP check. Good catch on that one.